src/utils/jws_utils/src/jws_utils.c

```diff
@@ -219,6 +219,7 @@ static bool ExtractJWSHeader(const char* jws, char** header)
     }
 
     strncpy(tempHeader, jws, headerLen);
```

Comment:
`src\utils\c_utils\src\string_c_utils.c`

```C
/**
 * @brief Allocates memory for @p target and copy @p len characters of @p source into @p target buffer.
 *
 * @param[out] target Output string
 * @param source Source string
 * @param len Length of string to copy.
 * @return bool Returns true is success.
 */
bool MallocAndSubstr(char** target, char* source, size_t len)
```

Replace all malloc/strncpy with calls to this rather than doing 1 offs.

and remove this in that code:

```C
    if (strncpy(t, source, len) != t)
    {
        free(t);
        return false;
    }
```

strncpy **cant** fail.  It returns source.

Having said all that, I'm not quite sure MallocAndSubstr is null-terminating either!

Reply:
this and other uses of strncpy will be addressed by removing its usage across the code base in develop branch.  The intent of this PR is for cherry-picking the low-risk fix of this particular issue.